procaddr.tmpl

```diff
@@ -0,0 +1,78 @@
+// This file implements GlowGetProcAddress for each platform. The correct
+// version is chosen automatically based on build tags:
+//
+// windows: WGL
+// darwin: CGL
+// linux: GLX
+//
+// Use of EGL instead of the platform's default (listed above) is made possible
+// via the "egl" build tag.
+//
+// It is also possible to install your own function outside this package for
+// retrieving OpenGL function pointers, to do this see InitWithProcAddrFunc.
+
+package {{.Name}}
+
+/*
+#cgo windows CFLAGS: -DTAG_WINDOWS
+#cgo windows LDFLAGS: -lopengl32
+
+#cgo darwin CFLAGS: -DTAG_DARWIN
+#cgo darwin LDFLAGS: -framework OpenGL
+
+#cgo linux CFLAGS: -DTAG_LINUX
+#cgo linux LDFLAGS: -lGL
+
+#cgo egl CFLAGS: -DTAG_EGL
+#cgo egl LDFLAGS: -lEGL
+
+#ifdef TAG_WINDOWS
+	#define WIN32_LEAN_AND_MEAN 1
+	#include <windows.h>
+	#include <stdlib.h>
+	static HMODULE ogl32dll = NULL;
+	void* GlowGetProcAddress(const char* name) {
+		void* pf = wglGetProcAddress((LPCSTR)name);
+		if(pf) {
+			return pf;
+		}
+		if(ogl32dll == NULL) {
+			ogl32dll = LoadLibraryA("opengl32.dll");
+		}
+		return GetProcAddress(ogl32dll, (LPCSTR)name);
+	}
+#endif
+
+#ifdef TAG_DARWIN
```

Comment:
You're correct on all accounts except the actual code you pasted (that's not valid C I think). It would look like:

```
#if defined(CREDIT)
    credit();
#elif defined(DEBIT)
    debit();
#elserom 
    printerror();
#endif
```

From [MSDN here](https://msdn.microsoft.com/en-us/library/ew2hz0yd.aspx).